plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+{ 
+    gazebo::msgs::Contacts contacts;
+    contacts = this->m_sensor->Contacts();
+    
+    ignition::math::Vector3d resultantForceonCG;
+    ignition::math::Vector3d resultantMomentonCG;
+        
+    double timestamp = -1;
+    for (size_t i = 0; i < contacts.contact_size(); i++)
+    {
+        gazebo::common::Time currentStamp;
+        for (int j=0; j < contacts.contact(i).position_size(); j++)
+        {
+            double t = contacts.contact(i).time().sec() + TIME_CONVERSION_NS*contacts.contact(i).time().nsec();
+            if (timestamp < 0)
+            {
+                timestamp = t;
+            }
+
+            // This block allows addition of wrench components only if they arrive in same timestamp
+            if (!ignition::math::equal(t, timestamp))
+                continue;
+
+            gazebo::msgs::JointWrench wrench = contacts.contact(i).wrench(j);
+
+            // we are interested only in the link specific contact wrenches and not the contact surface
+            if (wrench.body_1_name() == this->m_linkCollisionName)
+            {
+                resultantForceonCG += ignition::math::Vector3d(wrench.body_1_wrench().force().x(),
+                                                            wrench.body_1_wrench().force().y(),
+                                                            wrench.body_1_wrench().force().z());
+                resultantMomentonCG += ignition::math::Vector3d(wrench.body_1_wrench().torque().x(),
+                                                             wrench.body_1_wrench().torque().y(),
+                                                             wrench.body_1_wrench().torque().z());
+            }
+            else if (wrench.body_2_name() == this->m_linkCollisionName)
+            {
+                resultantForceonCG += ignition::math::Vector3d(wrench.body_2_wrench().force().x(),
+                                                            wrench.body_2_wrench().force().y(),
+                                                            wrench.body_2_wrench().force().z());
+                resultantMomentonCG += ignition::math::Vector3d(wrench.body_2_wrench().torque().x(),
+                                                             wrench.body_2_wrench().torque().y(),
+                                                             wrench.body_2_wrench().torque().z());
+            }
+        }
+    }
+
+#if DEBUG 
+     std::cout << "Wrench at CG [ " << resultantForceonCG.X() << " " <<  resultantForceonCG.Y() << " " << resultantForceonCG.Z() << " " << resultantMomentonCG.X() << " " <<  resultantMomentonCG.Y() << " " << resultantMomentonCG.Z() << " ]" << std::endl;
+#endif
+     
+    // Transform the wrench applied on the Link CG frame to the Link Origin Frame
+    // consider L: Link Origin Frame, C: Link CG Frame, ^: cross product operator givng a skew symmetric matrix
+    // L_f = L_R_C * C_f
```

Comment:
According to https://bitbucket.org/osrf/gazebo/issues/545/request-change-contact-reference-frame#comment-17039155 , the force and torques are already expressed with the orientation of the link frame (L). Are you sure that instead they are expressed in the orientation of the inertia matrix (C). Or perhaps in the link that we are testing L_R_C is the identity and for this reason everything works fine?

Reply:
http://osrf-distributions.s3.amazonaws.com/gazebo/api/dev/classgazebo_1_1sensors_1_1ContactSensor.html#a5e8c0561a71708862c4304d6174512e9

Here it says, The forces and torques are applied at the CG of perspective links for each collision body, specified in the inertial frame. That is why I took was using the transformation.

And yes, actually L_R_C in the current test case is identity.

Reply:
Upon a quick check, I can conclude two things: 
* In that documentation "inertial frame" was meant as "world frame", i.e. inertial frame according to Newton's first law, not "frame in which the inertia is expressed". 
* The documentation was originally added in 2013, but in 2015 the behavior of the function changed and the documentation was not updated.  

Details: 
* The documentation was originally added in https://bitbucket.org/osrf/gazebo/pull-requests/313/add-documentation-describing-data-returned (2013). At the same time the author of the PR opened an issue to return the data in "local frame": https://bitbucket.org/osrf/gazebo/issues/540/update-contactsensor-for-getting-contact . This probably means that at that time the contact data was not in a local frame. 
* The behavior of the function was changed in PR https://bitbucket.org/osrf/gazebo/pull-requests/355/bullet-contact-sensor (2015), but the documentation was never updated. 

I think it is safe to assume that those quantities are expressed with the orientation of the link frame. @prashanthr05 can you open an issue about this in the Gazebo bugtracker? Feel free to copy&paste part of this comment in the issue.